src/map/intif.c

```diff
@@ -336,31 +342,35 @@ static void intif_parse_account_storage(int fd)
 
 	Assert_retv(fd > 0);
 
-	payload_size = RFIFOW(fd, 2) - 8;
+	payload_size = RFIFOW(fd, 2) - 10;
 
 	if ((account_id = RFIFOL(fd, 4)) == 0 || (sd = map->id2sd(account_id)) == NULL) {
 		ShowError("intif_parse_account_storage: Session pointer was null for account id %d!\n", account_id);
 		return;
 	}
 
-	if (sd->storage.received == true) {
+	struct storage_data* stor = storage->ensure(sd, RFIFOW(fd, 8));
+	if (stor == NULL)
+		return;
+
+	if (stor->received == true) {
 		ShowError("intif_parse_account_storage: Multiple calls from the inter-server received.\n");
 		return;
 	}
 
 	storage_count = (payload_size/sizeof(struct item));
 
-	VECTOR_ENSURE(sd->storage.item, storage_count, 1);
+	VECTOR_ENSURE(stor->item, storage_count, 1);
```

Comment:
Memory manager reported a leak from this, I didn't check what is causing it (not sure if related to the missing VECTOR_CLEAR in unit.c

```
0001 : storage.c line 298 size 352 address 0x0x55976f3717e4
0002 : intif.c line 363 size 440 address 0x0x55977473f1fc
```